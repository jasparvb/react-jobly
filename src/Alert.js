import React from "react";

function Alert({ type, messages }) {
  return (
    <div className={`alert alert-${type}`} role="alert">
      {messages.map(err => (
        <p className="mb-0 small" key={err}>{err}</p>
      ))}
    </div>
  );
}

export default Alert;
