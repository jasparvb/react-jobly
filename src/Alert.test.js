import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";

it("renders without crashing", function() {
  render(<Alert />);
});

it("matches snapshot for danger", function() {
  let messages = ["Invalid credentials", "Not a valid URL"];
  const { asFragment } = render(<Alert type="danger" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for success", function() {
  let messages = ["Successfully logged in"];
  const { asFragment } = render(<Alert type="success" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});