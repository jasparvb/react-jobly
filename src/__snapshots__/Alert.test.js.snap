// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Invalid credentials
    </p>
    <p
      class="mb-0 small"
    >
      Not a valid URL
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Successfully logged in
    </p>
  </div>
</DocumentFragment>
`;
